test.js

```diff
@@ -459,9 +459,7 @@ test('odd file sizes', async t => {
 	const oddFileSizes = [1, 2, 3, 4, 5, 6, 7, 8, 10, 11, 12, 13, 255, 256, 257, 511, 512, 513];
 
 	for (const size of oddFileSizes) {
-		const buf = Buffer.alloc(size);
-		await t.notThrowsAsync(async () => {
-			await FileType.fromBuffer(buf);
-		}, `file size = ${size} bytes`);
+		const buffer = Buffer.alloc(size);
+		await t.notThrowsAsync(FileType.fromBuffer(buffer), `File size: ${size} bytes`);
```

Comment:
I have no idea why this is failing:

>  ✖ odd file sizes `t.notThrowsAsync()` must be called with a function or promise

`FileType.fromBuffer(buffer)` **is** a Promise.

Reply:
I am happy I am not the only one anymore who thinks the behavior of the async tests methods are weird. Passing a promise should be sufficient.

[test.js](https://github.com/sindresorhus/file-type/blob/master/test.js) is full of redundant `await` statements. 

I think in general async tests methods & await statements could be avoided by treating the test method acync if the return type of the test method is a promise.

Reply:
Ok, I know why it happens. When the file type cannot be detected, `.fromBuffer()` incorrectly returns `undefined` instead of `Promise<undefined>`.

Reply:
Aij

Reply:
It is missing `async` declaration:

https://github.com/sindresorhus/file-type/blob/e90035a1b46d3fe9347d90c8414ef9955c5f2701/core.js#L23

Reply:
I will create separate issue / PR for that. I am curious if the test cases are then working with a single await as well.

Reply:
And indeed, with the async fix, the async-test- methods do behave as expected :grin: